### PR TITLE
PIM-5853: Add number filter

### PR DESCRIPTION
**Pull request by willy-ahva**
<3 Thanks for taking the time to contribute! You're awesome! <3

If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/CONTRIBUTING.md 

**Description (for Contributor and Core Developer)**

What does this Pull Request do? reference the related issue?

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       |
| Added Behats                      |
| Changelog updated                 |
| Review and 2 GTM                  |
| Micro Demo to the PO (Story only) |
| Migration script                  |
| Tech Doc                          |

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/attribute/number.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    UserContext,
+    i18n,
+    template
+) {
+    return BaseFilter.extend({
+        shortname: 'number',
+        template: _.template(template),
+        events: {
+            'change [name="filter-operator"]': 'updateState',
+            'blur [name="filter-value"]': 'updateState'
```
Comment (willy-ahva) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/attribute/number.js]: try change

File: src/Pim/Bundle/EnrichBundle/Resources/config/form_extensions/filters.yml [hunk 2]
```diff
@@ -45,6 +45,19 @@ extensions:
                 - "<"
                 - EMPTY
 
+    pim-filter-attribute-number:
+        module: pim/filter/attribute/number
+        config:
+            operators:
```
Comment (willy-ahva) [inline src/Pim/Bundle/EnrichBundle/Resources/config/form_extensions/filters.yml]: check operators

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/attribute/number.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            return BaseFilter.prototype.initialize.apply(this, arguments);
+        },
+
+        /**
+         * {@inherit}
+         */
+        isEmpty: function () {
+            return !_.contains(['EMPTY', 'NOT EMPTY'], this.getOperator()) &&
+                (undefined === this.getValue() || '' === this.getValue())
+                ;
```
Comment (willy-ahva) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/attribute/number.js]: ;

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/attribute/number.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                operator: this.getOperator(),
+                operatorChoices: this.config.operators
+            }));
+        },
+
+        /**
+         * {@inheritdoc}
+         */
+        postRender: function () {
+            this.$('[name="filter-operator"]').select2({minimumResultsForSearch: -1});
```
Comment (willy-ahva) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/attribute/number.js]: .operator

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/attribute/number.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        },
+
+        /**
+         * {@inherit}
+         */
+        updateState: function () {
+            var operator = this.$('[name="filter-operator"]').val();
+            var value    = null;
+
+            if (!_.contains(['EMPTY', 'NOT EMPTY'], this.getOperator())) {
```
Comment (willy-ahva) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/attribute/number.js]: operator

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/attribute/number.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+                value = Number(this.$('[name="filter-value"]').val());
```
Comment (willy-ahva) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/attribute/number.js]: cats only if not ""

File: src/Pim/Bundle/EnrichBundle/Resources/config/form_extensions/filters.yml (hunk 2 above)
Comment (willy-ahva) [inline src/Pim/Bundle/EnrichBundle/Resources/config/form_extensions/filters.yml]: done

File: features/Pim/Behat/Decorator/Export/Filter/NumberDecorator.php [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        $operatorField = $this->decorate(
+            $this->find('css', '.select2-container.operator'),
+            ['Pim\Behat\Decorator\Field\Select2Decorator']
+        );
+        $operatorField->setValue($operator);
+
+        if (null !== $value && '' !== $value) {
+            $fieldValue = $this->spin(function () {
+                return $this->find('css', '.value');
+            }, 'Field value for number did not found.');
```
Comment (phaseinducer) [inline features/Pim/Behat/Decorator/Export/Filter/NumberDecorator.php]: -did

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/attribute/number.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        /**
+         * {@inherit}
+         */
+        updateState: function () {
+            var operator = this.$('[name="filter-operator"]').val();
+            var value    = null;
+
+            if (!_.contains(['EMPTY', 'NOT EMPTY'], operator)) {
+                value = this.$('[name="filter-value"]').val();
+                if (!_.isEmpty(value)) {
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/attribute/number.js]: Why `_.isEmpty()`? Your value is not an object.

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/filter/attribute/number.html [hunk 9]
```diff
@@ -0,0 +1,11 @@
+<select class="select2 operator" <%- editable ? '' : 'disabled' %> name="filter-operator">
+    <% _.each(operatorChoices, function (operatorChoice) { %>
+        <option value="<%- operatorChoice %>"<%- operator === operatorChoice ? ' selected' : '' %>>
+            <%- __('pim_enrich.export.product.filter.number.operators.' + operatorChoice) %>
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/filter/attribute/number.html]: Use `this.shortname` please.

**Pull request closed without merging**